tasks/internal/create-advisory-task/create-advisory-task.yaml

```
@@ -153,8 +157,13 @@ spec:
           # This also cds into the git repo
           git_clone_and_checkout --repository "$GIT_REPO" --revision "$REPO_BRANCH"
 
-          CONTENT_IMAGES=/tmp/content_images.json
-          jq -c '.content.images // []' <<< "$ADVISORY_JSON" > "$CONTENT_IMAGES"
+          spec_content_type=".content.images"
+          if [ "$(params.contentType)" != "image" ]; then
+            echo "Content type is generic artifact."
+            spec_content_type=".content.artifacts"
+          fi
+          CONTENT_FILE=/tmp/content.json
+          jq -c "${spec_content_type} // []" <<< "$ADVISORY_JSON" > "$CONTENT_FILE"
```

I know that we do this kind of thing for boolean values often (e.g. `if [ $param != true ]; then do something`), but I wonder if it would make sense to make sure it's really one of the two supported values here.
```suggestion
          if [ "$(params.contentType)" = "image" ]; then
            echo "Content type is image."
            spec_content_type=".content.images"
          elif [ "$(params.contentType)" = "generic" ]; then
            echo "Content type is generic artifact."
            spec_content_type=".content.artifacts"
          else
            echo "Unsupported contentType: $(params.contentType)"| tee -a "$STDERR_FILE"
            echo "Exiting." | tee -a "$STDERR_FILE"
            exit 1
          fi
          CONTENT_FILE=/tmp/content.json
          jq -c "${spec_content_type} // []" <<< "$ADVISORY_JSON" > "$CONTENT_FILE"
```

---

Implementing your suggestion with a slight tweak to include binary content type.

---

cool